src/BlockTemplatesController.php

```diff
@@ -785,98 +786,47 @@ protected function has_migrated_page( $page_id ) {
 	}
 
 	/**
-	 * Prepare default page template.
+	 * Stores an option to indicate that a template has been migrated.
 	 *
-	 * @param \WP_Post $page Page object.
-	 * @return string
+	 * @param string $page_id Page ID.
 	 */
-	protected function get_default_migrate_page_template( $page ) {
-		$default_template_content  = $this->get_block_template_part( 'header' );
-		$default_template_content .= '
-			<!-- wp:group {"layout":{"inherit":true}} -->
-			<div class="wp-block-group">
-				<!-- wp:heading {"level":1} -->
-				<h1 class="wp-block-heading">' . wp_kses_post( $page->post_title ) . '</h1>
-				<!-- /wp:heading -->
-				' . wp_kses_post( $page->post_content ) . '
-			</div>
-			<!-- /wp:group -->
-		';
-		$default_template_content .= $this->get_block_template_part( 'footer' );
-
-		return $default_template_content;
+	public function set_has_migrated_page( $page_id ) {
+		update_option( 'has_migrated_' . $page_id, 1 );
 	}
 
 	/**
 	 * Migrates a page to a template if needed.
 	 *
-	 * @param string   $page_id Page ID.
+	 * @param string   $template_slug Template slug.
 	 * @param \WP_Post $page Page object.
 	 */
-	protected function migrate_page( $page_id, $page ) {
-		if ( ! $page || empty( $page->post_content ) ) {
-			update_option( 'has_migrated_' . $page_id, '1' );
-			return;
-		}
+	protected function migrate_page( $template_slug, $page ) {
```

Comment:
Can we move this along the `set_migrated_page()` method into the `BlockTemplateMigrationUtils` class?